scripts/BLmotors

```diff
@@ -0,0 +1,311 @@
+
+
+def motorsInfo(motors, motorsName, cfg):
+    """
+    This function finds the real Pv name for each
+    record (alias), the cconfiguration name, and if
+    that configuration does matches or ot with the
+    actual configuration. Then it will create a list
+    per each record and finally it will be addes to
+    another list.
+    """
+    pm = pmgrAPI.pmgrAPI("ims_motor", cfg)
+    totalinfo = []
+    for motor in range(len(motors)):
+        infoperRecord = []
+        try:
+            motorPtwo = ophyd.signal.EpicsSignal(motorsName[motor]).get()
+            try:
+                mconf = pm.get_config(motors[motor])
+                exConfig = pm.diff_config(motors[motor])
+                if exConfig:
+                    infoperRecord.append(motors[motor])
+                    infoperRecord.append(motorPtwo)
+                    infoperRecord.append(mconf)
+                    infoperRecord.append("No")
+                else:
+                    infoperRecord.append(motors[motor])
+                    infoperRecord.append(motorPtwo)
+                    infoperRecord.append(mconf)
+                    infoperRecord.append("Yes")
+
+            except Exception:
+                infoperRecord.append(motors[motor])
+                infoperRecord.append(motorPtwo)
+                infoperRecord.append("None")
+                infoperRecord.append(" ")
+        except Exception:
+            try:
+                mconf = pm.get_config(motors[motor])
+                exConfig = pm.diff_config(motors[motor])
+                if exConfig:
+                    infoperRecord.append(motors[motor])
+                    infoperRecord.append("No found!")
+                    infoperRecord.append(mconf)
+                    infoperRecord.append("No")
+                else:
+                    infoperRecord.append(motors[motor])
+                    infoperRecord.append("No found!")
+                    infoperRecord.append(mconf)
+                    infoperRecord.append("Yes")
+            except Exception:
+                infoperRecord.append(motors[motor])
+                infoperRecord.append("No found!")
+                infoperRecord.append(" ")
+                infoperRecord.append(" ")
+        totalinfo.append(infoperRecord)
+    return totalinfo
+
+
+def displayfullInfo(totalinfo, argNoconf, argMatch,
+                    argFound, argsaliasConf, cfg):
+    """
+    This function displays in a table the information
+    of all the ims motors by selection of the user.
+    """
+    noConf = [info for info in totalinfo if info[0] == info[1]
+              and info[2] == "None" or info[0] != info[1]
+              and info[3] != "None" and info[1] != "No found!"]
+    noMatchConf = [info for info in totalinfo if info[3] == "No"]
+    noFound = [info for info in totalinfo if info[1] == "Not found!"]
+    aliasConf = [info for info in totalinfo if info[0] != info[1]
+                 and info[1] != "No found!" and info[2] != "None"]
```

Comment:
I suspect the above logic was difficult to debug. This is complicated enough to switch to using dictionaries or dataclasses so you can compare named fields instead of array indices, which would make this much easier to understand.